lib/ret_web/controllers/page_controller.ex

```diff
@@ -119,12 +121,77 @@ defmodule RetWeb.PageController do
     supports_pwa = ua.family != "Safari" && ua.family != "Mobile Safari"
 
     if supports_pwa do
-      conn |> render_asset("manifest.webmanifest")
+      manifest =
+        case Cachex.get(:assets, :manifest) do
+          {:ok, nil} ->
+            default_app_description =
+              "Share a virtual room with friends. Watch videos, play with 3D objects, or just hang out."
+
+            manifest =
+              Phoenix.View.render_to_string(RetWeb.PageView, "manifest.webmanifest",
+                app_name: get_app_config_value("translations|en|app-name") || "Hubs",
+                app_description: get_app_config_value("translations|en|app-description") || default_app_description,
+                app_icon: get_app_config_owned_file_uri("images|app_icon") || @default_app_icon
+              )
+
+            unless module_config(:skip_cache) do
+              Cachex.put(:assets, :manifest, manifest, ttl: :timer.seconds(15))
+            end
+
+            manifest
+
+          {:ok, manifest} ->
+            manifest
+        end
+
+      conn
+      |> put_resp_header("content-type", "application/manifest+json")
+      |> send_resp(200, manifest)
     else
       conn |> send_resp(404, "Not found.")
     end
   end
 
+  def render_for_path("/favicon.ico", _params, conn) do
+    favicon = get_configurable_asset(:app_config_favicon, "images|favicon", "favicon.ico")
+
+    conn
+    |> put_resp_header("content-type", "image/x-icon")
+    |> send_resp(200, favicon)
+  end
+
+  def render_for_path("/app-thumbnail.png", _params, conn) do
+    thumbnail = get_configurable_asset(:app_config_app_thumbnail, "images|app_thumbnail", "app-thumbnail.png")
+
+    conn
+    |> put_resp_header("content-type", "image/png")
+    |> send_resp(200, thumbnail)
+  end
+
+  defp get_configurable_asset(cache_key, config_key, fallback_file) do
+    case Cachex.get(:assets, cache_key) do
```

Comment:
any reason not to use `Cache.fetch` and use the fallback approach /w skip_cache  (similar to HEAD with app_config's caching)

Reply:
I wanted the cache to contain either the appconfig asset or the fallback asset from the page warmer.